k2/csrc/array.h

```diff
@@ -410,10 +417,46 @@ class Array2 {
     that
      it will have a different memory layout than the input.
   */
-  Array2<T> To(ContextPtr ctx) {
-    // TODO
-    Array2<T> array;
-    return array;
+  Array2<T> To(ContextPtr ctx) const {
+    if (dim0_ * dim1_ == 0) {
+      if (!region_) return *this;
+
+      if (ctx->IsCompatible(*Context())) return *this;
+
+      Array2 ans(ctx, dim0_, dim1_);
+      return ans;
+    }
+
+    if (ctx->IsCompatible(*Context()) && elem_stride0_ == dim1_) return *this;
```

Comment:
line break please

Reply:
... and fewer blank lines.

Reply:
fixed.

---

I prefer more blanks as I think it is more readable. I know fewer blank lines save screen spaces.